assets/js/main.js

```diff
@@ -296,23 +296,39 @@ function elasticSearch(query, callback) {
     searchReq.setRequestHeader('Authorization', `Bearer ${searchOnlyKey}`);
     searchReq.send(JSON.stringify(payload));
 }
+
+var searchSelectionId = 0;
+var searchListingLength = 0;
 function search() {
     var searchInput = $('.search-input');
     var searchButton = $('.search-button');
     var searchResult = $('.search-result');
+    var modalOverlay = $('.modal-overlay');
+    var body = $('body,html');
+    let focusOnFirst;
 
-    searchInput.on('keyup', function (e) {
+    searchInput.on('input', function (e) {
         elasticSearch(e.target.value, function () {
             var data = JSON.parse(this.responseText);
             var output = '';
-            data.results.forEach(function (post) {
+            data.results.forEach(function (post, index) {
                 output += `<div class="search-result-row">
-                        <a class="search-result-row-link" href="${post['url_path'].raw}">
+                        <a id="search-element-${index}" class="search-result-row-link" href="${post['url_path'].raw}">
                             ${post.title.raw}
                         </a>
                     </div>`;
             });
             searchResult.html(output);
+            searchListingLength = data.results.length;
+            searchSelectionId = -1;
+
+            clearTimeout(focusOnFirst);
+            focusOnFirst = setTimeout(function () {
+                if (searchListingLength == 0) return;
+                if (searchSelectionId >= 0) return;
```

Comment:
Should it be `searchSelectionId > 0` since we are trying to select the first element?

Reply:
the index of the first element is 0, hence should be >=0. searchSelectionId = -1 in this case represents the search box
As for the AdvisorySG entry, I think its an ElasticSearch (Backend Issue), so maybe it could be spun off into another issue